### PR TITLE
Warn if jvm api cannot be used

### PR DESCRIPTION
**Pull request by tpasternak**

### CODE REVIEW

**Comment by tpasternak** [comment]
image

File: modules/build/src/main/scala/scala/build/Build.scala [hunk 1]
```diff
@@ -480,16 +479,36 @@ object Build {
...
 
-  def prepareBuild(
+  def releaseFlag(
+    options: BuildOptions,
+    logger: Logger
+  ): Option[Int] = {
+    val bloopJvmV = options.javaOptions.bloopJvmVersion
+    val javaHome  = options.javaHome()
+    if (bloopJvmV.exists(javaHome.value.version > _.value)) {
+      logger.log(List(scala.build.errors.Diagnostic(
```
Comment (alexarchambault) [inline modules/build/src/main/scala/scala/build/Build.scala]: NIT
```suggestion
      logger.log(List(Diagnostic(
```

File: modules/build/src/main/scala/scala/build/ConsoleBloopBuildClient.scala [hunk 2]
```diff
@@ -204,4 +205,33 @@ object ConsoleBloopBuildClient {
...
+    positions: Seq[Position]
+  ): Unit = {
+    val isWarningOrError = true
+    if (isWarningOrError) {
+      val red    = Console.RED
+      val yellow = Console.YELLOW
+      val reset  = Console.RESET
+      val prefix =
+        if (severity == Severity.Error) s"[${red}error$reset] "
+        else s"[${yellow}warn$reset] "
```
Comment (alexarchambault) [inline modules/build/src/main/scala/scala/build/ConsoleBloopBuildClient.scala]: NIT Factor those as private fields in `object ConsoleBloopBuildClient`?

File: modules/build/src/main/scala/scala/build/ConsoleBloopBuildClient.scala [hunk 3]
```diff
@@ -204,4 +205,33 @@ object ConsoleBloopBuildClient {
...
+      val msgIt = message.linesIterator
+      out.println(s"${prefix}bloop:" + (if (msgIt.hasNext) " " + msgIt.next() else ""))
```
Comment (alexarchambault) [inline modules/build/src/main/scala/scala/build/ConsoleBloopBuildClient.scala]: Why a `"bloop"` here?

File: modules/build/src/main/scala/scala/build/ConsoleBloopBuildClient.scala [hunk 4]
```diff
@@ -204,4 +205,33 @@ object ConsoleBloopBuildClient {
...
+      for (line <- msgIt)
+        out.println(prefix + line)
+
+      for { pos <- positions } pos match {
```
Comment (alexarchambault) [inline modules/build/src/main/scala/scala/build/ConsoleBloopBuildClient.scala]: NIT Syntax
```suggestion
      positions.foreach {
```

File: modules/build/src/test/scala/scala/build/tests/BuildProjectTests.scala [hunk 5]
```diff
@@ -0,0 +1,135 @@
...
+      ),
+      javaOptions = JavaOptions(
+        bloopJvmVersion =
+          Some(Positioned(bloopJavaPath, bloopJvmVersion)),
+        javaHomeOpt = Some(Positioned.none(os.Path(javaHome)))
+      ),
+      scalaOptions = ScalaOptions(scalacOptions = scalacOptions)
+    )
+
+    val inputs  = new Inputs(Nil, None, os.pwd, "project", false)
```
Comment (alexarchambault) [inline modules/build/src/test/scala/scala/build/tests/BuildProjectTests.scala]: NIT
```suggestion
    val inputs  = Inputs(Nil, None, os.pwd, "project", false)
```

File: modules/build/src/test/scala/scala/build/tests/BuildProjectTests.scala [hunk 6]
```diff
@@ -0,0 +1,135 @@
...
+    val sources = Sources(Nil, Nil, None, Nil, options)
+    val logger  = new LoggerMock()
+    val res     = Build.buildProject(inputs, sources, Nil, options, Scope.Test, logger)
+
+    val scalaCompilerOptions = res.right.get.scalaCompiler.scalacOptions
+    (scalaCompilerOptions, res.right.get.javacOptions, logger.diagnostics)
+  }
+
+  def jvm(v: Int) = os.proc("cs", "java-home", "--jvm", s"zulu:$v").call().out.text().trim()
```
Comment (alexarchambault) [inline modules/build/src/test/scala/scala/build/tests/BuildProjectTests.scala]: Ideally, we should use the [same value as in the ITs](https://github.com/alexarchambault/scala-cli/blob/a14bb70db2fc2c5190d512632e112d3133013712/modules/integration/src/test/scala/scala/cli/integration/TestUtil.scala#L63) (just `"cs"` is a problem when the launcher is `cs.bat` on Windows, but not when it's `cs.exe`, IIRC).

File: modules/build/src/main/scala/scala/build/ConsoleBloopBuildClient.scala [hunk 7]
```diff
@@ -204,4 +208,29 @@ object ConsoleBloopBuildClient {
...
+      positions.foreach { pos =>
+        pos match {
+          case Position.Bloop(bloopJavaPath) =>
+            val bloopOutputPrefix = s"[current bloop jvm] "
+            out.println(prefix + bloopOutputPrefix + bloopJavaPath)
+            out.println(prefix + " " * bloopOutputPrefix.length + "^" * bloopJavaPath.length())
+          case pos => out.println(prefix + pos.render())
+        }
+      }
+    }
```
Comment (alexarchambault) [inline modules/build/src/main/scala/scala/build/ConsoleBloopBuildClient.scala]: NIT (more idiomatic)
```suggestion
      positions.foreach {
        case Position.Bloop(bloopJavaPath) =>
          val bloopOutputPrefix = s"[current bloop jvm] "
          out.println(prefix + bloopOutputPrefix + bloopJavaPath)
          out.println(prefix + " " * bloopOutputPrefix.length + "^" * bloopJavaPath.length())
        case pos => out.println(prefix + pos.render())
      }
    }
```

File: modules/build/src/test/scala/scala/build/tests/BuildProjectTests.scala [hunk 8]
```diff
@@ -0,0 +1,135 @@
...
+      ),
+      scalaOptions = ScalaOptions(scalacOptions = scalacOptions)
+    )
+
+    val inputs  = Inputs(Nil, None, os.pwd, "project", false)
+    val sources = Sources(Nil, Nil, None, Nil, options)
+    val logger  = new LoggerMock()
+    val res     = Build.buildProject(inputs, sources, Nil, options, Scope.Test, logger)
+
+    val scalaCompilerOptions = res.right.get.scalaCompiler.scalacOptions
```
Comment (alexarchambault) [inline modules/build/src/test/scala/scala/build/tests/BuildProjectTests.scala]: NIT (This avoids trapping the exception when there's one.)
```suggestion
    val scalaCompilerOptions = res.fold(throw _, identity).scalaCompiler.scalacOptions
```

File: modules/build/src/test/scala/scala/build/tests/BuildProjectTests.scala [hunk 9]
```diff
@@ -0,0 +1,135 @@
...
+    (scalaCompilerOptions, res.right.get.javacOptions, logger.diagnostics)
```
Comment (alexarchambault) [inline modules/build/src/test/scala/scala/build/tests/BuildProjectTests.scala]: NIT Same as just above
```suggestion
    (scalaCompilerOptions, res.fold(throw _, identity).javacOptions, logger.diagnostics)
```

**Review by alexarchambault — COMMENTED** [review]
👍 

Only left minor NIT comments.

**Review by alexarchambault — APPROVED** [review]

**Pull request merged**